### PR TITLE
 [MM-58500] Turn off PostedAck when the connection is no longer registered

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When the websocket connection is closed at the client, the server still keeps the connection object alive and will process `POSTED` events for that connection, even if they go nowhere. We cleanup the connections eventually, but in that time we are counting notifiable events still that will never get an ACK.

This PR flips the `PostedAck` flag off on the connection at unregistry so that we don't expect ACKs back for the POSTED events that we send. This should improve the metrics a bit since we were adding measuring errors here.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-58500

```release-note
NONE
```

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]

**Review by agnivade — CHANGES_REQUESTED** [review]

**Review by agnivade — COMMENTED** [review]
Actually, wondering if it would be better to simply check for webconn.active in broadcast_hooks.go? That should make the logic simpler. In this PR, you are effectively using the `active` state and the `PostedAck` state to do the same thing.

Something like:

```go
if !webConn.PostedAck || !webConn.active.Load(){
	return nil
}
```

Although you would need to write a method to expose the active flag. That shouldn't be a big deal.

**Comment by devinbinnie** [comment]
> Actually, wondering if it would be better to simply check for webconn.active in broadcast_hooks.go? That should make the logic simpler. In this PR, you are effectively using the `active` state and the `PostedAck` state to do the same thing.
> 
> Something like:
> 
> ```go
> if !webConn.PostedAck || !webConn.active.Load(){
> 	return nil
> }
> ```
> 
> Although you would need to write a method to expose the active flag. That shouldn't be a big deal.

This was actually my preferred solution, but since `active` wasn't exposed I didn't want to go through all the trouble of exposing it. However, given the above comments I think it makes sense to do so as well, so let's do that.

**Comment by devinbinnie** [comment]
@amyblais Any chance I can get this into v9.9? This is a measuring error for the notification metrics.

**Comment by amyblais** [comment]
> @amyblais Any chance I can get this into v9.9? This is a measuring error for the notification metrics.

QA is currently wrapping up testing for v9.9.0 today, so I think we can take this into v9.9.1 once it gets scheduled.

File: server/channels/app/platform/web_conn.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ type WebConn struct {
 	deadQueuePointer int
 	// active indicates whether there is an open websocket connection attached
```
Comment (agnivade) [inline server/channels/app/platform/web_conn.go:108]: nit: `Active`. Now that it's exported.

**Review by agnivade — APPROVED** [review]

File: server/channels/app/platform/web_conn.go (hunk 1 above)
Comment (devinbinnie) [reply server/channels/app/platform/web_conn.go:108]: Sorry, auto-merge beat me before I could turn it off :P

Comment (agnivade) [reply server/channels/app/platform/web_conn.go:108]: I was wondering how did you merge the PR one second after I approved it :P

**Pull request merged**